graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -54,42 +126,46 @@
 ]
 RenameTypesT = TypeVar("RenameTypesT", bound=RenameTypes)
 # AST visitor functions can return a number of different things, such as returning a Node (to update
-# that node) or returning a special value defined in graphql.visitor such as REMOVE (to remove the
-# node) and IDLE (to do nothing with the node). In the current GraphQL-core version (>=3,<3.1),
-# REMOVE is set to the singleton object Ellipsis and IDLE is set to None. However, because these
-# special values' underlying definitions can change, we can't type-hint functions returning a
-# special value with anything more specific than Any. For more information, see:
-# https://github.com/kensho-technologies/graphql-compiler/pull/834#discussion_r434622400
-# We can update this type hint when a future GraphQL-core release organizes these special return
-# values into an enum.
-# https://github.com/graphql-python/graphql-core/issues/96
-VisitorReturnType = Union[Node, Any]
+# that node) or returning a special value specified in graphql.visitor's VisitorAction.
+VisitorReturnType = Union[Node, VisitorAction]
 
 
 def rename_schema(
-    schema_ast: DocumentNode, renamings: Mapping[str, str]
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]]
 ) -> RenamedSchemaDescriptor:
     """Create a RenamedSchemaDescriptor; types and query type fields are renamed using renamings.
 
     Any type, interface, enum, or fields of the root type/query type whose name
-    appears in renamings will be renamed to the corresponding value. Any such names that do not
-    appear in renamings will be unchanged. Scalars, directives, enum values, and fields not
-    belonging to the root/query type will never be renamed.
+    appears in renamings will be renamed to the corresponding value if the value is not None. If the
+    value is None, it will be suppressed in the renamed schema and queries will not be able to
+    access it.
+
+    Any such names that do not appear in renamings will be unchanged.
+
+    Scalars and directives not belonging to the root/query type will never be renamed. In addition,
```

Comment:
Confused about this sentence. It's a bit ambiguous: it can be parsed as "(scalars and directives) (not belonging to...)" or as "(scalars) and (directives not belonging to...)", and neither seems the clear meaning.

Also, I'm not sure what "belonging to the root/query type" means here. I don't know how scalars or directives can be seen as belonging to anything, and I'm not sure we'll ever want to rename directives.

Reply:
I re-read the original PR (#390) and I think I mis-parsed the original comment in the way you described in the first paragraph. When I read `Scalars, directives, enum values, and fields not belonging to the root/query type will never be renamed` in the context of everywhere else it mentions scalars, directives, or enum values, the correct reading in the original code is "(Scalars), (directives), (enum values), and (fields not belonging to the root/query type)". In this case, "field not belonging to the root/query type" should actually be "type that exists in the schema but not as a field in the object type `SchemaQueryType`".

I'll update this to only mention directives, since renaming/suppressing scalars types looks to be very similar to doing so with object types, which we've already implemented here.